### PR TITLE
[x86/Linux] Fix compile error in codeman

### PR DESCRIPTION
**Pull request by seanshpark**
Fix compile error for x86/Linux
- add !_TARGET_X86_ to WIN64EXCEPTIONS directives
- fix "use of undeclared identifier 'NEED_TO_PORT_THIS_ONE'"
- fix "no member named 'GetUnwindInfo' in '_hpCodeHdr'"
- fix "use of undeclared identifier 'RUNTIME_FUNCTION__EndAddress'"

### CODE REVIEW

**Comment by RussKeldorph** [comment]
Please ignore the x86 compatjit and x86 legacy_backend jobs if they fail.  They have been removed from future PRs.

File: src/vm/i386/cgencpu.h [hunk 1]
```diff
@@ -87,6 +87,10 @@ BOOL Runtime_Test_For_SSE2();
 // Needed for PInvoke inlining in ngened images
 #define HAS_NDIRECT_IMPORT_PRECODE              1
 
+#ifdef FEATURE_PAL
+#define USE_INDIRECT_CODEHEADER
```
Comment (jkotas) [inline src/vm/i386/cgencpu.h]: This should be left undefined (see my comment about changing the ABI in the other PR).

File: src/vm/codeman.cpp [hunk 2]
```diff
@@ -1720,7 +1726,11 @@ BOOL EEJitManager::LoadJIT()
         {
             // Now, load the compat jit and initialize it.
 
+#if !defined(FEATURE_PAL)
             LPCWSTR pwzJitName = MAKEDLLNAME_W(L"compatjit");
+#else
+            LPCWSTR pwzJitName = MAKEDLLNAME_W(u"compatjit");
```
Comment (jkotas) [inline src/vm/codeman.cpp]: There is `W(...)` macro to do this without ifdefs.

File: src/vm/i386/cgencpu.h (hunk 1 above)
Comment (seanshpark) [inline src/vm/i386/cgencpu.h]: OK, thank you for point this.

**Comment by seanshpark** [comment]
If I tried to block `LazyGetFunctionEntry` function(s) , it lead to no where, so just block the function body not to use `GetUnwindInfo()` and fix it properly after build is ready.

**Comment by jkotas** [comment]
Pretty much all `defined(WIN64EXCEPTIONS)` in this file should be `defined(WIN64EXCEPTIONS) && !defined(_TARGET_X86_)` - similar to what you have done under src/zap.

**Comment by jkotas** [comment]
Maybe we should invent a new #define for this.

**Comment by seanshpark** [comment]
>  invent a new #define

Good to hear :) . Codes seems not about Exception. Could you give a good new directive?

> Pretty much all `defined(WIN64EXCEPTIONS)` in this file should be `defined(WIN64EXCEPTIONS) && !defined(_TARGET_X86_)`

The change spread to several files, threads.cpp, jitinterface.cpp, eedbginterfaceimpl.cpp and h, stackwalk.cpp and h and maybe so on. It is scary to change not knowing much.

@parjong has also tried fixing this compile error another way so if adding `_ASSERTE(!"Fix This");` is inappropriate, I'll change this patch only leaving with `W`macro change.

**Comment by jkotas** [comment]
> It is scary to change not knowing much.

You may be better off to just make it compile on Linux x86 same way as it compile on Windows x86 (ie without WIN64EXCEPTIONS defined). It should allow you to get hello world running, with much fewer changes. And then try to turn on WIN64EXCEPTIONS as isolated change that is not mixed up with the rest.

**Comment by seanshpark** [comment]
@ OK Thanks, I'll try again :)

**Comment by seanshpark** [comment]
Please wait, need to resolve with another PR by parjong.

**Comment by janvorli** [comment]
> You may be better off to just make it compile on Linux x86 same way as it compile on Windows x86 (ie without WIN64EXCEPTIONS defined).

Yeah, I am starting to think the same. I have not realized before that the WIN64EXCEPTIONS is used at that many places in the source.

**Comment by seanshpark** [comment]
> without WIN64EXCEPTIONS defined

Yes, It's getting huge. I'll close this PR and restart from without `WIN64EXCEPTIONS`

**Pull request closed without merging**